static/js/protein/chart.ts

```diff
@@ -16,147 +16,659 @@
-    return x.name;
+  // converts the tissue score value to string label
+  function formatTick(d) {
+    return TISSUE_SCORE_TO_LABEL[d];
+  }
+  // dictionary mapping tissues to specific colors
+
+  // groups the tissues of a similar origin and sorts them in ascending order
+  reformattedData.sort(function (x, y) {
+    const a = TISSUE_COLOR_DICT[x.name];
+    const b = TISSUE_COLOR_DICT[y.name];
+    if (a < b) {
+      return -1;
+    }
+    if (a > b) {
+      return 1;
+    }
+    if (a === b) {
+      if (x.value < y.value) {
+        return -1;
+      } else {
+        return 1;
+      }
+    }
   });
 
-  const height = HEIGHT - MARGIN.top - MARGIN.bottom;
-  const width = WIDTH - MARGIN.left - MARGIN.right;
+  // specifying graph specific dimensions
+  const height = 200 - MARGIN.top - MARGIN.bottom;
+  const width = 860 - MARGIN.left - MARGIN.right;
+
+  const svg = d3
+    .select("#tissue-score-chart")
+    .append("svg")
+    .attr("width", width + MARGIN.left + MARGIN.right)
+    .attr("height", height + MARGIN.top + MARGIN.bottom)
+    .append("g")
+    .attr("transform", "translate(" + MARGIN.left + "," + MARGIN.top + ")");
+  // plots x-axis for the graph - tissue names
   const x = d3
-    .scaleLinear()
-    .domain([0, d3.max(reformattedData, (d) => d.value)])
-    .range([MARGIN.left, width - MARGIN.right]);
-  const y = d3
     .scaleBand()
-    .domain(arrName)
-    .rangeRound([MARGIN.top, height - MARGIN.bottom])
-    .padding(0.1);
-  function formatTick(d) {
-    return TISSUE_SCORE_TO_LABEL[d];
+    .range([0, width])
+    .domain(
+      reformattedData.map(function (d) {
+        return d.name;
+      })
+    )
+    .padding(0.15);
+  svg
+    .append("g")
+    .attr("transform", "translate(0," + height + ")")
+    .call(d3.axisBottom(x))
+    .selectAll("text")
+    .attr("transform", "translate(-10,0)rotate(-45)")
+    .style("text-anchor", "end");
+  // y-axis for the graph - tissue score
+  const y = d3.scaleLinear().domain([0, 3]).range([height, 0]);
+  svg.append("g").call(d3.axisLeft(y).ticks(4).tickFormat(formatTick));
+  // plotting the bars
+  svg
+    .selectAll("tissue-score-bar")
+    .data(reformattedData)
+    .enter()
+    .append("rect")
+    .attr("x", (d) => x(d.name))
+    .attr("y", (d) => y(d.value))
+    .attr("height", function (d) {
+      return height - y(d.value);
+    })
+    .attr("width", x.bandwidth())
+    .style("fill", function (d) {
+      return TISSUE_COLOR_DICT[d.name];
+    })
+    .on("mouseover", function () {
+      d3.select(this).transition().duration(50).style("opacity", 0.5);
+    })
+    .on("mouseout", function () {
+      d3.select(this).transition().duration(50).style("opacity", 1);
+    });
+}
+
+export function drawProteinInteractionChart(
+  id: string,
+  data: InteractingProteinType[]
+): void {
+  // retrieves the parent protein name
+  // TODO: create a helper function for reformatting
+  const parentProtein = data[1].parent;
```

Comment:
curious why you're looking at the second element in data & if this was intended, you should check that data has at least 2 elements

Reply:
You're right, it should be 0 instead of 1. Here, basically all the elements will have the same parent protein name so I'm retrieving the parent protein name using the first element only.